healthcare/api-client/v1/hl7v2/hl7v2_messages_test.py

```diff
@@ -0,0 +1,191 @@
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.get_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    hl7v2_messages.delete_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    out, _ = capsys.readouterr()
+
+    # Check that create/get/list/delete worked
+    assert 'Created HL7v2 message' in out
+    assert 'Name' in out
+    assert 'Deleted HL7v2 message' in out
+
+
+def test_ingest_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.ingest_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.get_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    hl7v2_messages.delete_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    out, _ = capsys.readouterr()
+
+    # Check that ingest/get/list/delete worked
+    assert 'Ingested HL7v2 message' in out
+    assert 'Name' in out
+    assert 'Deleted HL7v2 message' in out
+
+
+def test_patch_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.create_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.patch_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id,
+        label_key,
+        label_value)
+
+    hl7v2_messages.delete_hl7v2_message(
```

Comment:
The same concern about cleaning up.

If the API doesn't allow recursive deletion, one idea is to list the messages and delete them in the clean up code.

Reply:
Does it matter if we're deleting the HL7v2 store anyway? Deleting the store deletes any messages inside of it.

Reply:
If the API behaves like that, it's fine.

However, note that it will be not OK once the API changes the behavior (not allowing recursive deletion).